tree/20_cfe_basics/log_rudder.cf

```diff
@@ -930,3 +930,34 @@ bundle agent method_id_pop()
     pass3.debug::
       "${configuration.debug} Popping method id ${method_id} from method stack (length: ${stack_len}, new length: ${new_len})";
 }
+
+bundle agent _rudder_report_state(class_prefix, message, state)
+{
+  # We cannot allow for empty component_name
+  vars:
+      "component_key" string => "${report_data.component_key}";
+
+      "final_state" string => "${state}";
+    pass1.!report::
+      "final_state" string => "log_info", if => "${class_prefix}_noop.!${class_prefix}_kept.!${class_prefix}_repaired.!${class_prefix}_error";
+      "final_state" string => "log_info", if => "${class_prefix}_kept.!${class_prefix}_repaired.!${class_prefix}_error";
```

Comment:
this can be simplified into 
"final_state" string => "log_info", if => (${class_prefix}_noop|"${class_prefix}_kept).!${class_prefix}_repaired.!${class_prefix}_error";